client/client.go

```diff
@@ -192,7 +196,17 @@ func (c *httpClusterClient) reset(eps []string) error {
 		neps[i] = *u
 	}
 
+	// TODO: rebalance when new endpoint appears
+	npinned := -1
```

Comment:
I would suggest that randomly choosing one endpoint when reseting for now. We can do better later including the TODO.